packages/mapper/src/mapper.ts

```diff
@@ -140,32 +141,22 @@ export default class Mapper {
     const db = collection._database._name;
     const coll = collection._name;
 
-    const dbOptions: any = {};
+    const dbOptions: DatabaseOptions = {};
 
     if ('readConcern' in options) {
       dbOptions.readConcern = options.readConcern;
     }
     if ('writeConcern' in options) {
-      dbOptions.writeConcern = options.writeConcern;
+      Object.assign(dbOptions, options.writeConcern);
     }
 
-    let cmd;
-    if (coll === null) {
```

Comment:
This would be unreachable code, a collection always have a name and is always passed as parameter by shell api.